pubsub/api/Pubsub.Samples/ResumePublish.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START pubsub_quickstart_resume_publish]
+// [START pubsub_publish]
+
+using Google.Cloud.PubSub.V1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class ResumePublishSample
+{
+    public async Task<int> PublishOrderedMessagesAsync(string projectId, string topicId, IEnumerable<KeyValuePair<string, string>> messagesAndKeys)
+    {
+        TopicName topicName = TopicName.FromProjectTopic(projectId, topicId);
+
+        var customSettings = new PublisherClient.Settings
+        {
+            EnableMessageOrdering = true
+        };
+
+        PublisherClient publisher = await PublisherClient.CreateAsync(topicName, settings: customSettings);
+
+        int publishedMessageCount = 0;
+        var publishTasks = messagesAndKeys.Select(async messageAndKey =>
+        {
+            try
+            {
+                string message = await publisher.PublishAsync(messageAndKey.Value, messageAndKey.Key);
+                Console.WriteLine($"Published message {message}");
+                Interlocked.Increment(ref publishedMessageCount);
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine($"An error occurred when publishing message {messageAndKey.Key}: {exception.Message}");
+                publisher.ResumePublish(messageAndKey.Value);
```

Comment:
You should also increment the counter after this line.

Reply:
But in this case, the publish actually failed. Would it still make sense to increment the counter?

Reply:
But resume publish will probably succeed? But it's fine, you are counting success on first try, and I was thinking on success overall.